.github/workflows/build.yaml

```diff
@@ -179,7 +179,7 @@ jobs:
       - name: Setup Go 1.21
         uses: actions/setup-go@v5
         with:
-          go-version: '1.21'
+          go-version: "1.21"
```

Comment:
This should probably be 1.23 as well.

Reply:
We can also determine the go version to use in CI from the `go.mod` file and forget about inconsistencies in the future: https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file

Reply:
TIL Thanks!

Reply:
Updated this place and made use of `go-version-file` everywhere